egs/avspeech/model.py

```diff
@@ -0,0 +1,381 @@
+
+        #Batch normalization layers
+
+        self.batch_norm1 = nn.BatchNorm2d(96)
+        self.batch_norm2 = nn.BatchNorm2d(96)
+        self.batch_norm3 = nn.BatchNorm2d(96)
+        self.batch_norm4 = nn.BatchNorm2d(96)
+        self.batch_norm5 = nn.BatchNorm2d(96)
+        self.batch_norm6 = nn.BatchNorm2d(96)
+        self.batch_norm7 = nn.BatchNorm2d(96)
+        self.batch_norm8 = nn.BatchNorm2d(96)
+        self.batch_norm9 = nn.BatchNorm2d(96)
+        self.batch_norm10 = nn.BatchNorm2d(96)
+        self.batch_norm11 = nn.BatchNorm2d(96)
+        self.batch_norm11 = nn.BatchNorm2d(96)
+        self.batch_norm12 = nn.BatchNorm2d(96)
+        self.batch_norm13 = nn.BatchNorm2d(96)
+        self.batch_norm14 = nn.BatchNorm2d(96)
+        self.batch_norm15 = nn.BatchNorm2d(8)
+
+    def get_padding(self,kernel_size,dilation):
+        padding = (((dilation[0])*(kernel_size[0]-1))//2,((dilation[1])*(kernel_size[1]-1))//2)
+        return padding
+
+    def forward(self,input_audio):
+        # input audio will be (2,298,257)
+
+        output_layer = F.relu(self.batch_norm1(self.conv1(input_audio)))
+        #print (output_layer.shape)
+        output_layer = F.relu(self.batch_norm2(self.conv2(output_layer)))
+        #print (output_layer.shape)
+        output_layer = F.relu(self.batch_norm3(self.conv3(output_layer)))
+        #print (output_layer.shape)
+        output_layer = F.relu(self.batch_norm4(self.conv4(output_layer)))
+        #print (output_layer.shape)
+        output_layer = F.relu(self.batch_norm5(self.conv5(output_layer)))
+        #print (output_layer.shape)
+        output_layer = F.relu(self.batch_norm6(self.conv6(output_layer)))
+        #print (output_layer.shape)
+        output_layer = F.relu(self.batch_norm7(self.conv7(output_layer)))
+        #print (output_layer.shape)
+        output_layer = F.relu(self.batch_norm8(self.conv8(output_layer)))
+        #print (output_layer.shape)
+        output_layer = F.relu(self.batch_norm9(self.conv9(output_layer)))
+        #print (output_layer.shape)
+        output_layer = F.relu(self.batch_norm10(self.conv10(output_layer)))
+        #print (output_layer.shape)
+        output_layer = F.relu(self.batch_norm11(self.conv11(output_layer)))
+        #print (output_layer.shape)
+        output_layer = F.relu(self.batch_norm12(self.conv12(output_layer)))
+        #print (output_layer.shape)
+        output_layer = F.relu(self.batch_norm13(self.conv13(output_layer)))
+        output_layer = F.relu(self.batch_norm14(self.conv14(output_layer)))
+        output_layer = F.relu(self.batch_norm15(self.conv15(output_layer)))
+
+        # output_layer will be (N,8,298,257)
+        # we want it to be (N,8*257,298,1)
+        batch_size = output_layer.size(0) # N
+        height = output_layer.size(2)  # 298
+
+        output_layer = output_layer.view(batch_size,-1,height,1)
```

Comment:
```
# output_layer will be (N,8,298,257)
# we want it to be (N,8*257,298,1)
```
This will mix the time and freq axis, you need to transpose before I guess. You should check Row and column major ordering. 
I might be wrong but I think there is a mistake here.

Reply:
Oh, good point!! I did not notice that!

Referring to this [answer](https://discuss.pytorch.org/t/whats-the-best-way-to-concatenate-these-pytorch-dimensions/35737/2).

Something like this should solve:
`output_layer.permute(0, 1, 3, 2).contiguous().view(batch_size, -1, height, 1)`

What do you think?

Reply:
That would work. Personally I would use `transpose` and `reshape`. 
```python 
output_layer.transpose(-1, -2).reshape((batch_size, -1, height, 1)
```
Using `transpose` makes that you have to read two axis instead of 4, it's easier to see the intent IMO. 
`reshape` is `view` if the resulting array is contiguous, otherwise it's `.contiguous().view()`

Reply:
Good to know, thanks!